UI-Dropdown: Use `v-select` option instead of `v-combox`

arturv2000 opened the pull request:
## Description

This allows the user to use the vue component `v-select` instead of `v-combox`.

The main difference is that `v-select` is not "writable". Specifies that the list is displayed by clicking the down arrow and that the text portion is not editable. This means that the user cannot enter a new value. Only values already in the list can be selected

This prevents that the user can type an invalid value (that do not exist on the list and sending payload as null as result). It allows typing for filtering values.

The text description in the config menu could be better, but having no ideia how. Other option is to use the v-select instead of v-combox, but this could be tricky for legacy reasons.

## Related Issue(s)

<!-- What issue does this PR relate to? -->

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

joepavitt commented:
I'd actually go further here - and replace `combobox` with `select` entirely as an iteration/ If I see requests come in for being able to define custom values in this, then we can re-introduce the `combobox` as an option, but surface it as "allow custom input", rather than exposing any concept of Vuetify components to end-users at the Node-RED level

arturv2000 commented:
> I'd actually go further here - and replace `combobox` with `select` entirely as an iteration.

Ok, I can make that change, and completely replace with `v-select`.


> If we see requests come in for being able to define custom values in this, then we can re-introduce the `combobox` as an option, but surface it as "allow custom input", rather than exposing any concept of Vuetify components to end-users at the Node-RED level

Actually, for that to work more modifications are necessary, as it is now, the user can "write" any value, and it will be present on the UI but it is not pass to node-red.

joepavitt commented:
> Actually, for that to work more modifications are necessary, as it is now, the user can "write" any value, and it will be present on the UI but it is not pass to node-red.

Even more reason for us to switch over to `v-select` then 😄

joepavitt commented:
Wait, actually, I just tried the `v-select`, the `v-combobox` I do prefer as it allows you to search your options (if you have hundreds this is _very_ useful)

It doesn't allow definition of a new option, and won't submit with custom values.

In which case, if you do have value is disabling the "search" - let's brand it that way. Have an option surfaced in Node-RED config which is "Allow Search" - enabled by default. If toggled off, we use the standard `v-select`

arturv2000 commented:
`v-select` allows search, but is necessary to write the the value correct from "start", if we have items that start with ´L´ and type `L` it will go to the first `L` start item in the list, but it does not `filter` the probable values like `combobox`.

Maybe the option should be called `Allow Search with Filter`?

Ii is a pity that `combobox` doesn't have a option like `Disable input values not in List` or a error indication if the value is not in the list.

joepavitt commented:
Ah yes, I do see that now, I think I prefer `combobox` for the moment as the default. I think an "Allow Search" is a good starting point (true by default), and we can iterate from there.

arturv2000 commented:
Updated...

Also added the prop [auto-select-first](https://vuetifyjs.com/en/components/combobox/#props-auto-select-first) when using combo-box, to update the selection (send message to node-red) based on item highlighted / selected via search.

arturv2000 commented:
Would say yes.

If deselecting the `Allow Search` it will use the `v-select` component that do not allow direct text input

colinl commented:
OK, I will test it when it has been released.
